internal/ratelimit/ratelimit.go

```diff
@@ -0,0 +1,260 @@
+package ratelimit
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/go-resty/resty/v2"
+)
+
+type RateLimitInfo struct {
+	Limit     int // x-ratelimit-limit
+	Period    int // x-ratelimit-period
+	Remaining int // x-ratelimit-remaining
+	Reset     int // x-ratelimit-reset (seconds until reset)
+}
+
+func (r *RateLimitInfo) IsNearLimit(threshold float64) bool {
+	if r.Limit == 0 {
+		return false
+	}
+	return float64(r.Remaining)/float64(r.Limit) < threshold
+}
+
+func (r *RateLimitInfo) ShouldThrottle(threshold float64) bool {
+	return r.IsNearLimit(threshold)
+}
```

Comment:
What is the difference between `IsNearLimit` and `ShouldThrottle`?

This seems like the same, as such I would prefer keeping only `ShouldThrottle`.

Reply:
Fixed